cloudflare/resource_cloudflare_load_balancer.go

```diff
@@ -0,0 +1,335 @@
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudFlareLoadBalancer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudFlareLoadBalancerCreate,
+		Read:   resourceCloudFlareLoadBalancerRead,
+		Update: resourceCloudFlareLoadBalancerUpdate,
+		Delete: resourceCloudFlareLoadBalancerDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudFlareLoadBalancerImport,
+		},
+
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"zone_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false, // allow reusing same load balancer for different dns names
+			},
+
+			"fallback_pool_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 32),
+			},
+
+			"default_pool_ids": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringLenBetween(1, 32),
+				},
+			},
+
+			"proxied": {
+				Type:          schema.TypeBool,
+				Optional:      true,
+				Default:       false,
+				ConflictsWith: []string{"ttl"},
+			},
+
+			"ttl": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"proxied"}, // this is set to zero regardless of config when proxied=true
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 1024),
+			},
+
+			// nb enterprise only
+			"pop_pools": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"pop": {
+							Type:     schema.TypeString,
+							Required: true,
+							// let the api handle validating pops
+						},
+
+						"pool_ids": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validation.StringLenBetween(1, 32),
+							},
+						},
+					},
+				},
+				Set: HashByMapKey("pop"),
```

Comment:
Is there a specific need for a custom hash function? `Set` should not be set unless there are very specific requirements to the hashing, otherwise please omit `Set` on any `TypeList`,`TypeMap`, or `TypeSet`, and let Terraform apply the default hashing for each type.

Reply:
I want to enforce uniqueness on the pop field (point-of-presence name) , I thought this was the best way to do it?